RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc

```diff
@@ -0,0 +1,239 @@
+        for (unsigned i = 0; i < prep_params.var_names.size(); ++i) {
+          const auto &varname = prep_params.var_names[i];
+          const auto &raw_value = taginfo.features().get(varname);
+          const auto &info = prep_params.info(varname);
+          int insize = center_norm_pad(raw_value,
+                                       info.center,
+                                       info.norm_factor,
+                                       prep_params.min_length,
+                                       prep_params.max_length,
+                                       vdata,
+                                       curr_pos,
+                                       info.pad,
+                                       info.replace_inf_value,
+                                       info.lower_bound,
+                                       info.upper_bound);
+          curr_pos += insize;
+          if (i == 0 && (!input_shapes_.empty())) {
+            input_shapes_[igroup][2] = insize;
+          }
+
+          if (debug_) {
+            std::cout << " -- var=" << varname << ", center=" << info.center << ", scale=" << info.norm_factor
+                      << ", replace=" << info.replace_inf_value << ", pad=" << info.pad << std::endl;
+            for (unsigned i = curr_pos - insize; i < curr_pos; i++) {
+              std::cout << vdata[i] << ",";
+            }
+            std::cout << std::endl;
+          }
+        }
+      }
+      input.toServer(tdata);
+    }
+  }
+}
+
+void ParticleNetSonicJetTagsProducer::produce(edm::Event &iEvent,
+                                              const edm::EventSetup &iSetup,
+                                              Output const &iOutput) {
+  edm::Handle<TagInfoCollection> tag_infos;
+  iEvent.getByToken(src_, tag_infos);
+
+  // initialize output collection
+  std::vector<std::unique_ptr<JetTagCollection>> output_tags;
+  if (!tag_infos->empty()) {
+    auto jet_ref = tag_infos->begin()->jet();
+    auto ref2prod = edm::makeRefToBaseProdFrom(jet_ref, iEvent);
+    for (std::size_t i = 0; i < flav_names_.size(); i++) {
+      output_tags.emplace_back(std::make_unique<JetTagCollection>(ref2prod));
+    }
+  } else {
+    for (std::size_t i = 0; i < flav_names_.size(); i++) {
+      output_tags.emplace_back(std::make_unique<JetTagCollection>());
+    }
+  }
+
+  if (!tag_infos->empty()) {
+    const auto &output1 = iOutput.begin()->second;
+    const auto &outputs_from_server = output1.fromServer<float>();
+
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      const auto &taginfo = (*tag_infos)[jet_n];
+
+      if (!taginfo.features().empty()) {
+        const auto &jet_ref = tag_infos->at(jet_n).jet();
+        for (std::size_t flav_n = 0; flav_n < flav_names_.size(); flav_n++) {
+          (*(output_tags[flav_n]))[jet_ref] = outputs_from_server[jet_n][flav_n];
+        }
+      } else {
+        const auto &jet_ref = tag_infos->at(jet_n).jet();
```

Comment:
jet_ref is exactly the same here and above, so it could be declared just once, outside of the if/else block

Reply:
Changed